lib/jekyll/filters.rb

```diff
@@ -260,6 +261,23 @@ def where_exp(input, variable, expression)
       end
     end
 
+    # Convert the input into integer
+    #
+    # input - the object string
+    #
+    # Returns the integer value
+    def to_integer(input)
+      return input if input.is_a?(Integer)
+      return 1 if input == true
+      return 0 if input == false
+      begin
+        input.to_i
+      rescue
+        raise ArgumentError,
```

Comment:
There is nothing ambiguous about that error.  Our only job is to do what we are told to do, we can either be defensive or we can throw, if we choose to throw then the native error will suffice, they told us to run to_i, and to_i doesn't exist, therefore the native error is the most literally right, we need not rescue that as it was exactly what happened, if we choose to be defensive then return 0 like Ruby does with unpareseable strings.